Documentation/teaching/labs/interrupts.rst

```diff
@@ -1024,3 +1080,39 @@ Reset the buffer if the device is written to. For this step follow the
 sections marked with **TODO 5** in the skeleton.
 
 Implement :c:func:`reset_buffer` and add the write operation to *kbd_fops*.
+
+.. attention:: In the write function Use :c:func:`spin_lock_irqsave` and
+         :c:func:`spin_unlock_irqrestore` for locking when resetting the
+         buffer.
+
+         Revisit the `Locking`_ section.
+
+For testing, you will need to create the */dev/kbd* character device
+driver using the mknod before reading from it. The device master and
+minor are defined as ``KBD_MAJOR`` and ``KBD_MINOR``:
+
+.. code-block:: c
+
+   mknod /dev/kbd c 42 0
+
+Build, copy and boot the virtual machine and load the module.
+Test it using the command:
+
+.. code-block:: c
+
+   cat /dev/kbd
+
+Press some keys, then run the command :command:`echo “clear” > /dev/kbd`.
+Check the buffer's content again. It should be reset.
+
+Extra Exercises
+===============
+
+1. kfifo
+---------
+
+Implement a keylogger using the
+`kfifo API <https://elixir.bootlin.com/linux/v4.15/source/include/linux/kfifo.h>`_.
+
+.. hint:: Follow the `API call examples from the kernel code <https://elixir.bootlin.com/linux/v4.15/source/samples/kfifo>`_.
+  For example, the file `bytestream-examples.c<https://elixir.bootlin.com/linux/v4.15/source/samples/kfifo/bytestream-example.c>`_.
```

Comment:
Add a space before `<`, otherwise the link is not displayed correctly. Check the generated page.

Reply:
Fixed.